Fixes #17611 - Localize smart variables strings

dLobatog opened the pull request:
This commit rename removes the prefixes of LookupKey from model
attributes so that we can reuse the same strings across smart class
params and smart variables. In addition to that it uses the label
attribute to ensure those are used

domcleal commented on app/views/lookup_keys/_fields.html.erb:
```diff
@@ -4,9 +4,9 @@
   <fieldset>
     <h2><%= _("Parameter details") %></h2>
-    <%= text_f f, :key, :disabled => is_param, :size => "col-md-8" %>
+    <%= text_f f, :key, :disabled => is_param, :size => "col-md-8", :label => _('Key') %>
```
No need to extract the strings again here, or even translate them, the form helpers do it automatically. They're likely to get out of sync.

domcleal commented on locale/model_attributes.rb:
```diff
@@ -267,58 +267,34 @@
 _('Variable lookup key')
 # TRANSLATORS: "Table name" or "Table name|Column name" for error messages
-_('VariableLookupKey|Avoid duplicates')
+_('Avoid duplicates')
```
These should be `Table name|Colum name` per the comment, i.e. `LookupKey|Avoid duplicates` or they won't match the automatic string lookups.

dLobatog commented:
I've updated it according to your comments @domcleal  - but I'm probably missing something. If I do *not* use `:label` on the view, I don't see any translations. That's because PuppetclassLookupKey and VariableLookupKey are used still on the .po files. Should I change those too?

domcleal commented:
If you want to test it you could change them temporarily, but they shouldn't be changed directly - they would be updated when the POT is updated and synced from Transifex.

You appear to have squashed your changes into an unrelated commit.

dLobatog committed:
Fixes #17611 - Localize smart variables strings

This commit rename removes the prefixes of LookupKey from model
attributes so that we can reuse the same strings across smart class
params and smart variables.

dLobatog commented:
Fair enough - thanks @domcleal !

domcleal commented:
Merged, thanks @dLobatog.
